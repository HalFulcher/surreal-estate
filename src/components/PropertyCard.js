import React, { useEffect } from "react";

const PropertyCard = ({
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
}) => {
  return (
    <div className="propertyDetails">
      <p>{title}</p>
      <p>{type}</p>
      <p>{bathrooms}</p>
      <p>{bedrooms}</p>
      <p>{price}</p>
      <p>{city}</p>
      <p>{email}</p>
    </div>
  );
};

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bathrooms: PropTypes.number.isRequired,
  bedrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};

export default PropertyCard;
