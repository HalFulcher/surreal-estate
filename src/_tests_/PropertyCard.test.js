import React from "react";
import { render, getByText, getByTestId } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "Property",
    type: "Flat",
    bathrooms: "3",
    bedrooms: "1",
    price: "100000",
    city: "Manchester",
    email: "test@gmail.com",
  };

  it("displays component", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays title property correctly", () => {
    const { asFragment } = render(<PropertyCard title={validProps.title} />);

    expect(getByText(/Property Name/).textContent).toBe("Property Name");

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays type property correctly", () => {
    const { asFragment } = render(<PropertyCard type={validProps.type} />);

    expect(getByText(/Flat/).textContent).toBe("Flat");

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays type property correctly", () => {
    const { asFragment } = render(
      <PropertyCard bathrooms={validProps.bathrooms} />
    );

    expect(getByTestId(/Flat/).textContent).toBe("Flat");

    expect(asFragment()).toMatchSnapshot();
  });
});
