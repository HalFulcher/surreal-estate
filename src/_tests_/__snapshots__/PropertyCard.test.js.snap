// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard displays component 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="property-card-title"
    >
      Property Title
    </div>
    <div
      class="property-card-type"
    >
      Flat
    </div>
    <div
      class="property-card-bathrooms"
      data-testid="property-card-bathrooms"
    >
      3
    </div>
    <div
      class="property-card-bedrooms"
      data-testid="property-card-bedrooms"
    >
      1
    </div>
    <div
      class="property-card-price"
      data-testid="property-card-price"
    >
      100000
    </div>
    <div
      class="property-card-city"
    >
      Manchester
    </div>
    <div
      class="property-card-email"
    >
      <a
        href="mailto:test@gmail.com"
      >
        test@gmail.com
      </a>
    </div>
  </div>
</DocumentFragment>
`;
